### DISCUSSION TITLE
About the expression of the phase_gradient_reflector helper function

### DISCUSSION THREAD

**Discussion by ekilcioglu** [NVlabs/sionna#503] [Q&A] [answered]
Hello,

I was working on the RIS support of the Sionna RT and realized something regarding the explanation of the phase_gradient_reflector helper function. I have reviewed both literature papers on phase gradient reflector and focusing lens approaches.

In the explanation of the phase_gradient_reflector helper function, the phase gradient is expressed with a minus sign and the subtraction is given as (k_i - k_r). However, based on my calculations, we should either remove the minus sign or change the subtraction to (k_r - k_i) while keeping the minus sign.

Is there a mistake in this expression, or did I make an error in my calculations? I reviewed the source code of this function, and it seems that the minus sign is not taken into account in the source code, so the helper function works properly.

Thank you in advance for your reply.

Best regards,
Emre

**Comment by ekilcioglu** [comment]
Up

**Reply by jhoydis** [reply, marked as answer]
Hi @ekilcioglu,

Good catch. That is simply a typo in the documentation. It will be fixed in the next release.

Thanks!
